lib/charms/grafana_k8s/v0/grafana_source.py

```diff
@@ -724,9 +724,21 @@ def _set_default_data(self) -> None:
 
     def set_peer_data(self, key: str, data: Any) -> None:
         """Put information into the peer data bucket instead of `StoredState`."""
-        self._charm.peers.data[self._charm.app][key] = json.dumps(data)  # type: ignore[attr-defined]
+        peers = self._charm.peers  # type: ignore[attr-defined]
+        if not peers:
+            logger.info("no peer relation: skipping set_peer_data call")
+            return
+
+        peers.data[self._charm.app][key] = json.dumps(data)  # type: ignore[attr-defined]
 
     def get_peer_data(self, key: str) -> Any:
         """Retrieve information from the peer data bucket instead of `StoredState`."""
+        peers = self._charm.peers  # type: ignore[attr-defined]
+        if not peers:
+            logger.warning(
```

Comment:
Should we link to the juju issue?
```suggestion
            # https://bugs.launchpad.net/juju/+bug/1998282
            logger.warning(
```